addons/relate/ftests/src/main/java/org/commonjava/indy/relate/ftest/RelDownloadPomNotFound404Test.java

```diff
@@ -0,0 +1,65 @@
+package org.commonjava.indy.relate.ftest;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.commonjava.indy.ftest.core.AbstractIndyFunctionalTest;
+import org.commonjava.indy.model.core.RemoteRepository;
+import org.commonjava.test.http.expect.ExpectationServer;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.InputStream;
+
+import static org.commonjava.indy.model.core.StoreType.remote;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Created by ruhan on 2/17/17.
+ */
+public class RelDownloadPomNotFound404Test
+                extends AbstractIndyFunctionalTest
+{
+    private static final String path = "org/foo/bar/1/bar-1.pom";
+
+    private static final String pathRel = path + ".rel";
+
+    private static final String content =
+                    "<project><modelVersion>4.0.0</modelVersion><groupId>org.foo</groupId><artifactId>bar</artifactId><version>1</version></project>";
+
+    @Rule
+    public ExpectationServer server = new ExpectationServer();
+
+    @Test
+    public void run() throws Exception
+    {
+        final String repo1 = "repo1";
+
+        server.expect( server.formatUrl( repo1, path ), 404, "not found" );
+
+        RemoteRepository remote1 = new RemoteRepository( repo1, server.formatUrl( repo1 ) );
+        client.stores().create( remote1, "adding remote", RemoteRepository.class );
+
+        // Download .rel before even touching POM
+        InputStream rel = client.content().get( remote, repo1, pathRel );
+        assertThat( rel, nullValue() );
+
+        logger.debug( ">>> register again ..." );
+        server.expect( server.formatUrl( repo1, path ), 200, content ); // upstream restored
```

Comment:
The first request that results in a 404 would normally log that in the not-found cache so the next request for it wouldn't open a connection to the remote server (faster that way). The not-found cache entry would then expire after some time and subsequent requests for it would result in Indy retrying the request to the remote server. I'm surprised this didn't happen for your request, but maybe it's because you're using a ContentGenerator, and this bypasses the NFC. But I don't think that's true, since the NFC is used in TransferManagerImpl, which is used both for ContentManager and DirectContentAccess, by way of DownloadManager.